From e88f3c4d5db9c359aae36754aae3504af6a5cf15 Mon Sep 17 00:00:00 2001
From: Sam Mortimer <sam@mortimer.me.uk>
Date: Fri, 19 Jan 2018 13:10:29 -0800
Subject: [PATCH 2/2] surfaceflinger: set a prop when initialization is
 complete

*) Set service.sf.primary_display_ready = 1 after we've a received
   a hotplug event for the primary display.

*) If hwcomposer starts before surfaceflinger, display hal sdm
   hwc_session.cpp doesn't initialize properly (running within
   surfaceflinger).  With this commit, init scripts can then
   "wait_for_prop service.sf.primary_display_ready 1" before
   starting hwcomposer to eliminate the race condition.

*) For ref, waiting on "init.svc.surfaceflinger running" is not good
   enough - sf is not far enough into it's initialization sequence.

Change-Id: I1cd4cf4ff50734ba1909a0ed61b46267e744f1c8
---
 services/surfaceflinger/SurfaceFlinger.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index d757746..42aad08 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -1317,6 +1317,7 @@ void SurfaceFlinger::onHotplugReceived(int32_t sequenceId,
             createBuiltinDisplayLocked(DisplayDevice::DISPLAY_PRIMARY);
         }
         createDefaultDisplayDevice();
+        property_set("service.sf.primary_display_ready", "1");
     } else {
         if (sequenceId != mComposerSequenceId) {
             return;
-- 
2.7.4

